[mercedesme] Additional mappings and channels

weymann opened the pull request:
Fixes #17266 

- **Mappings added**
   - service#starterBattery
   - tires#sensor-available
   - tires#marker*
  
- **Channels added** 
  see [readme](https://github.com/weymann/MercedesMe-OH-Drops/blob/main/README.md)
  - `eco` group *(new)*
    -  `accel` - Acceleration Score
    - `coasting`- Coasting Score
    - `constant` - Constant Driving Score
    - `bonus` - Bonus range towards *very sportive driver*

  - `hvac` group
    - `ac-status` - Status of preconditioning command
    - `aux-status` - Status of auxiliary heating command

  - `charge` group
    - `status` - Status of charging
    - `error` - In case of problem description of error 

  - `position` group
    - `status` - Positioning status

lsiepel commented on bundles/org.openhab.binding.mercedesme/src/main/resources/OH-INF/thing/eco-channel-types.xml:
```diff
@@ -0,0 +1,26 @@
+	<channel-type id="bonus">
+		<item-type>Number:Length</item-type>
+		<label>Bonus Range</label>
+		<state pattern="%d km" readOnly="true"/>
```
```suggestion
		<state pattern="%d %unit%" readOnly="true"/>
```

Doublecheck: when updating the channel, the default is m, make sure it is converted properly.

weymann replied:
Very good hint. Forgot to add channel to `MetaDataAdjuster`.

lsiepel commented on bundles/org.openhab.binding.mercedesme/src/main/resources/OH-INF/thing/hvac-channel-types.xml:
```diff
@@ -10,14 +10,28 @@
 	<channel-type id="temperature-hvac">
 		<item-type>Number:Temperature</item-type>
-		<label>AC Temperature</label>
+		<label>AC Desired Temperature</label>
```
```suggestion
		<label>AC Setpoint</label>
```
(for your consideration)

weymann replied:
True - changed to `AC Setpoint Temperature`

lsiepel reviewed (changes requested):
Overall LGTM. Very minor comments.
The thing upgrade instructions are missing for the eco group. the other new and changed channels are added dynamically so they don;t need instructions.

weymann commented:
> The thing upgrade instructions are missing for the eco group. the other new and changed channels are added dynamically so they don;t need instructions.

Added `instructions.xml` in `OH-INF/update` for added and updated channels: Please double check because till now I wasn't familiar with that.

lsiepel commented:
> > The thing upgrade instructions are missing for the eco group. the other new and changed channels are added dynamically so they don;t need instructions.
> 
> Added `instructions.xml` in `OH-INF/update` for added and updated channels: Please double check because till now I wasn't familiar with that.

They are fine. Only thing to add is that you also need to set the proiperty `thingTypeVersion` for each thing in their xml structure.
```
		<properties>
			<property name="thingTypeVersion">1</property>
		</properties>
````

weymann commented:
> > > The thing upgrade instructions are missing for the eco group. the other new and changed channels are added dynamically so they don;t need instructions.
> > 
> > 
> > Added `instructions.xml` in `OH-INF/update` for added and updated channels: Please double check because till now I wasn't familiar with that.
> 
> They are fine. Only thing to add is that you also need to set the proiperty `thingTypeVersion` for each thing in their xml structure.
> 
> ```
> 		<properties>
> 			<property name="thingTypeVersion">1</property>
> 		</properties>
> 
I really need some help here. [Documentation available](https://www.openhab.org/docs/developer/bindings/thing-xml.html#updating-thing-types) is not really covering all possible problems so I am  guessing around what's the right instructions.
I've a similar problem with https://github.com/openhab/openhab-addons/issues/17265 where I try out different solutions but the tutorial above doesn't provide any hint to solve it.

lsiepel commented:
As you managed to fix the other PR, can you fix the conflict and proceed or do you need some specific information to continue. Let me now an i'll do the best i can.

weymann commented:
> As you managed to fix the other PR, can you fix the conflict and proceed or do you need some specific information to continue. Let me now an i'll do the best i can.

Yes, looks fine now. Increased `thingTypeVersion` for due to new instructions, testing successful.
I'm good to go!

lsiepel reviewed (approved):
LGTM
